agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -204,7 +181,19 @@ def start(info)
       attach_router unless interface_ip('weave')
       connect_peers(peer_ips)
       info "using trusted subnets: #{trusted_subnets.join(',')}" if trusted_subnets && !already_started?
+      # Start the weavemesh plugin
+      plugin = nil
+      until plugin && plugin.running? do
+        exec_params = [
+          '--local', 'launch-plugin'
+        ]
+        self.exec(exec_params)
+        plugin = Docker::Container.get('weaveplugin') rescue nil
+        wait = Time.now.to_f + 10.0
+        sleep 0.5 until (plugin && plugin.running?) || (wait < Time.now.to_f)
 
+      end
+      ensure_kontena_network
```

Comment:
The order is wrong here. We can't `docker network create kontena` before we have etcd running, and etcd needs the `10.81.0.X` IP configured in the `post_start` -> `weave expose` below.